contracts/0.8.25/vaults/StakingVault.sol

```diff
@@ -313,27 +318,55 @@ contract StakingVault is IStakingVault, OwnableUpgradeable {
      * @param _deposits Array of deposit structs
      * @dev Includes a check to ensure StakingVault is balanced before making deposits
      */
-    function depositToBeaconChain(Deposit[] calldata _deposits) external {
+    function depositToBeaconChain(
+        Deposit[] calldata _deposits,
+        bytes32 _expectedGlobalDepositRoot,
+        bytes calldata _signature
+    ) external {
         if (_deposits.length == 0) revert ZeroArgument("_deposits");
-        ERC7201Storage storage $ = _getStorage();
 
+        bytes32 currentGlobalDepositRoot = BEACON_CHAIN_DEPOSIT_CONTRACT.get_deposit_root();
+        if (_expectedGlobalDepositRoot != currentGlobalDepositRoot)
+            revert GlobalDepositRootMismatch(_expectedGlobalDepositRoot, currentGlobalDepositRoot);
+
+        ERC7201Storage storage $ = _getStorage();
         if (msg.sender != $.nodeOperator) revert NotAuthorized("depositToBeaconChain", msg.sender);
         if ($.beaconChainDepositsPaused) revert BeaconChainDepositsArePaused();
         if (!isBalanced()) revert Unbalanced();
 
         uint256 totalAmount = 0;
         uint256 numberOfDeposits = _deposits.length;
+        // XOR is a commutative operation, so the aggregate root will be the same regardless of the order of deposits
+        bytes32 depositDataBatchXorRoot;
+
         for (uint256 i = 0; i < numberOfDeposits; i++) {
```

Comment:
I'd rather see here two loops and the signature validation before the deposit if we don't save the huge amount of gas here.